CHANGELOG.md

```diff
@@ -1,5 +1,229 @@
+
+    ```toml
+    webpack_config: webpack.prod.js
+    ```
+
+    [EverlastingBugstopper]: https://github.com/EverlastingBugstopper
+    [issue/246]: https://github.com/cloudflare/wrangler/issue/246
+    [pull/253]: https://github.com/cloudflare/wrangler/pull/253
+
+  - **Enforce one Webpack entry in configuration - [xtuc], [pull/245]**
+
+     `wrangler` now returns an error during the build process if you use a webpack configuration with more than one export – `wrangler` needs to have a single known export from webpack to know what to build!
+
+    [xtuc]: https://github.com/xtuc
+    [pull/245]: https://github.com/cloudflare/wrangler/pull/245
+
+  - **Add user agent - [xtuc], [issue/234] [pull/236]**
+
+    For every outgoing request, `wrangler` includes a `User-Agent` header to clearly indicate to servers and APIs that a `wrangler` client is making a request: `wrangler/dev` in debug mode and `wrangler/$version` in release mode.
+
+    [xtuc]: https://github.com/xtuc
+    [issue/234]: https://github.com/cloudflare/wrangler/issue/234
+    [pull/236]: https://github.com/cloudflare/wrangler/pull/236
+
+  - **Display commands in their defined order - [Electroid], [pull/236]**
+
+    We've re-arranged the order of the commands when you run `wrangler` without any subcommands, so that commonly-used commands are easier to find!
+
+    [Electroid]: https://github.com/Electroid
+    [pull/233]: https://github.com/cloudflare/wrangler/pull/236
+
+  - **Show sizes - [xtuc], [pull/205]**
+
+    Once the build is finished, `wrangler` now prints the compressed size of the script, and, if available, the Wasm binary size.
+
+    [xtuc]: https://github.com/xtuc
+    [pull/205]: https://github.com/cloudflare/wrangler/pull/205
+
+  - **Add HTTP prefix to publish command output. - [elithrar], [pull/198]**
+
+    Prefix "https://" in front of the "script available" output to allow shells to automatically detect it as a link. Many shells will allow you to click directly on the URL from inside the terminal (such as iTerm's "CMD-Click"), making it much easier to navigate to your subdomains, or any published Workers applications.
+
+    [elithrar]: https://github.com/elithrar
+    [pull/198]: https://github.com/cloudflare/wrangler/pull/198
+
+  - **Build: add message and emoji - [xtuc], [pull/193]**
+
+    The `wrangler` team _really_ loves emoji, so we made sure to send a little bit of ✨ cheer ✨ your way, via a new message and emoji, whenever you use the `build` subcommand. 💌
+
+    [xtuc]: https://github.com/xtuc
+    [pull/193]: https://github.com/cloudflare/wrangler/pull/193
+
+- ### 🤕 Fixes
+
+  - **Use rust template for rust project - [EverlastingBugstopper], [pull/309]**
+
+  The `--type` argument when generating a new Wrangler project previously was unable to handle a provided `rust` type. This PR adds support for the `rust` type, and uses the [rustwasm-worker-template](https://github.com/cloudflare/rustwasm-worker-template) when generating a new project.
```

Comment:
Previously when passing `—type rust` to `wrangler generate`, the only indication that it worked was that the `type` in `wrangler.toml` was `rust`. There were no Rust files in the default template for a `rust` type project. Now, when running `wrangler generate —type rust`, wrangler will use [rustwasm-worker-template](https://github.com/cloudflare/rustwasm-worker-template).

Reply:
Maybe wording here needs changed a bit, but it’s inaccurate to say that wrangler was “unable to handle a provided `rust` type.. it could handle it, it just didn’t produce a project with a relevant template

Reply:
oh, sure - thanks